Added additional documentations and references for how CustomField, ComputedField, and Relationship behave in Nautobot v2.0. (#4137)
nautobot/nautobot: nautobot/docs/user-guide/platform-functionality/relationship.md

To create a relationship, from the top-level navigation menu select **Extensibility > Data Management > Relationships**

!!! tip

>     Because relationship information can be included in the REST API and in GraphQL, when defining a relationship, you must provide a `key` that contains underscores rather than dashes (`my_relationship_key`, not `my-relationship-key`), as some features would not work optimally if dashes are included in the key. The `key` attribute is derived from its `label` attribute (unique for all Relationship instances). The key of a Relationship instance is also used as the key in the underlying relationship data dictionary.

Reviewer: ```suggestion
    Because relationship information can be included in the REST API and in GraphQL, when defining a relationship, you must provide a `key` that contains underscores rather than dashes (`my_relationship_key`, not `my-relationship-key`), as some features would not work optimally if dashes are included in the key. If not explicitly specified, the `key` attribute is derived from the `label` attribute (unique for all Relationship instances).
```

Relationship connections can be labeled with a friendly name so that when they are displayed in the GUI, they will have a more descriptive or friendly name.
From the Devices/Circuits example above, you might label the relationship so that on the Device side the connection appears as 'Terminated Circuits' and on the Circuit side the connection appears as 'Terminating Devices'.

## Relationship Keys

Relationship model's `key` attribute is derived from its `label` attribute (unique for all Relationship instances). The key of a Relationship instance contains a GraphQL-safe string used as the key in the underlying relationship data dictionary and it can be used to retrieve data from the API and GraphQL.

> 

Reviewer: Would this flow more logically if added adjacent to the note at lines 7-8 of this file, or even added as a component of that note?
